src/app.js

```diff
@@ -93,15 +93,20 @@ class App {
     return url.parse(cozyUrl)
   }
 
-  // Return a promise for registering a device on the remote cozy
-  registerRemote (cozyUrl, redirectURI, onRegistered, deviceName) {
+  // Check that the cozyUrl is valid
+  checkCozyUrl (cozyUrl) {
     let parsed = this.parseCozyUrl(cozyUrl)
     cozyUrl = url.format(parsed)
     if (!['http:', 'https:'].includes(parsed.protocol) || !parsed.hostname) {
       let err = new Error(`Your URL looks invalid: ${cozyUrl}`)
       log.warn(err)
       throw err
     }
+    return cozyUrl
+  }
+
+  // Return a promise for registering a device on the remote cozy
+  registerRemote (cozyUrl, redirectURI, onRegistered, deviceName) {
```

Comment:
Except for the config change above, couldn't we call `checkCozyUrl` in `registerRemote`?
It could even be done in `Registration` since `checkCozyUrl` (and `parseCozyUrl`) doesn't really depend on `App`.